x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java

```diff
@@ -88,17 +88,17 @@ public final void setDocument(int docId) {
     }
 
     /**
-     * Expose the {@code _source} to the script.
+     * Expose field data to the script as {@code doc}.
      */
-    protected final Map<String, Object> getSource() {
-        return leafSearchLookup.source();
+    public final Map<String, ScriptDocValues<?>> getDoc() {
+        return leafSearchLookup.doc();
     }
 
     /**
-     * Expose field data to the script as {@code doc}.
+     * A non-public convenience method that's used in testing.
      */
-    public final Map<String, ScriptDocValues<?>> getDoc() {
-        return leafSearchLookup.doc();
+    protected final Map<String, Object> getSource() {
+        return leafSearchLookup.source();
     }
```

Comment:
Do we actually need this method at all?  All the tests that call it also have access to the `leafSearchLookup` object so they could just use that?

Reply:
you mean that we should make leafSearchLookup protected? I don't think the tests have access to it today but maybe they should.

Reply:
Right, `leafSearchLookup` is currently private. I was just thinking of this as a tiny incremental improvement to avoid another person being confused by the javadoc.

Reply:
I mean the tests currently look like this:
```
return (fieldName, params, lookup) -> (ctx) -> new BooleanFieldScript(fieldName, params, lookup, ctx) {
                                    @Override
                                    public void execute() {
                                        for (Object foo : (List<?>) getSource().get("foo")) {
                                            emit((Boolean) foo);
                                        }
                                    }
                                };
```
where `lookup` is passed as a parameter to the lambda, so we could rewrite them like this:
```
return (fieldName, params, lookup) -> (ctx) -> new BooleanFieldScript(fieldName, params, lookup, ctx) {
                                    @Override
                                    public void execute() {
                                        for (Object foo : (List<?>) lookup.source().get("foo")) {
                                            emit((Boolean) foo);
                                        }
                                    }
                                };
```

Reply:
Thanks @romseygeek for the suggestion, I updated the PR to remove `AbstractFieldScript#getSource` entirely.

Reply:
Makes sense to me too, thanks